### ISSUE TITLE
Only show Release Notes warnings if unavailable fields are being accessed

### ISSUE THREAD

**Issue by Bio2hazard** [rfennell/vNextBuild#95]
### VSTS Extension you are using
*Generate Release Notes*

### Where are you running it?
*VSTS*

### Version of Extension/Task
*2.5.1*

### Expected behavior and actual behavior
Expected: Not giving a bunch of warnings that are irrelevant to a release based on a GitHub build.

Actual: Giving a bunch of warnings that are irrelevant to a release based on a GitHub build.

I think the warnings should only be produced if a `csdetail` field is requested that is unavailable on github.

### Steps to reproduce the problem
1. Add a Build that triggers off of a Github repository
2. Add a Release that triggers off of the build from step 1
3. Add Generate Release Notes step with Github template

I am getting the following warnings:
> Unable to get details of changeset/commit as it is not stored in TFS/VSTS
> For [commit hash] location [github link]
> Just using the details we have from the build

I am using the following template Inline:
```
# Release notes for release $($release.name)
**Release completed** $("{0:dd/MM/yy HH:mm:ss}" -f [datetime]$release.modifiedOn) 

## Builds  
@@BUILDLOOP@@
### $($build.definition.name)  
**Build Number**  : $($build.buildnumber)    
**Build completed** $("{0:dd/MM/yy HH:mm:ss}" -f [datetime]$build.finishTime)     
**Source Branch** $($build.sourceBranch)  
  
#### Associated commits  
@@CSLOOP@@  
* **ID $($csdetail.id)** [$($csdetail.message)]($($csdetail.displayUri)) on $("{0:dd/MM/yy HH:mm:ss}" -f [datetime]$csdetail.timestamp)
@@CSLOOP@@  


----------

@@BUILDLOOP@@
```

Thank you for the great Task!

**Comment by rfennell** [comment]
So if I understand you correctly the issue is that the logging is too verbose, saying the same thing twice, and I suppose it is making an extra call to get csdetails we actually know will fail?

But the actual output release notes are OK?

My initial thought is not to do anything about this, trading the extra failed try/catch I have now against having to pass more state with an extra if test.

Or am I missing your point?

**Comment by Bio2hazard** [comment]
![image](https://cloud.githubusercontent.com/assets/2462038/23808647/6ba41ac2-057f-11e7-88a2-bc4b7819529f.png)

The issue is that my release has warning messages that I can't disable and that are irrelevant to me. I know I use github, so I use the github template and only access `csdetail.id` , `csdetail.message` , `csdetail.displayUri` and `csdetail.timestamp` which all populate normally. 

There should not be a warning unless I try to access data only available in TFS/VSTS.

I just don't get the point of the warning. As far as I can tell I am not doing anything wrong, and moving all my code from GitHub to VSTS isn't an option, and the release notes are working fine. It seems like an unactionable warning that will just perpetually spam my release summary for no reason.

**Comment by rfennell** [comment]
Yes I see your point, warning is the wrong type, they are not build warnings, they are logging issues. I will have a look at the logging type I am using so the first message is in the standard build log, and maybe even drop the other messages, at least make them verbose

**Comment by rfennell** [comment]
Fix released in 2.6.2
